server/mailet/integration-testing/src/test/java/org/apache/james/transport/mailets/amqp/AmqpRule.java

```diff
@@ -0,0 +1,115 @@
+/****************************************************************
+ * Licensed to the Apache Software Foundation (ASF) under one   *
+ * or more contributor license agreements.  See the NOTICE file *
+ * distributed with this work for additional information        *
+ * regarding copyright ownership.  The ASF licenses this file   *
+ * to you under the Apache License, Version 2.0 (the            *
+ * "License"); you may not use this file except in compliance   *
+ * with the License.  You may obtain a copy of the License at   *
+ *                                                              *
+ *   http://www.apache.org/licenses/LICENSE-2.0                 *
+ *                                                              *
+ * Unless required by applicable law or agreed to in writing,   *
+ * software distributed under the License is distributed on an  *
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY       *
+ * KIND, either express or implied.  See the License for the    *
+ * specific language governing permissions and limitations      *
+ * under the License.                                           *
+ ****************************************************************/
+
+package org.apache.james.transport.mailets.amqp;
+
+import java.io.IOException;
+import java.net.InetAddress;
+import java.util.Optional;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.TimeoutException;
+
+import org.apache.james.util.streams.SwarmGenericContainer;
+import org.junit.rules.ExternalResource;
+
+import com.google.common.base.Charsets;
+import com.google.common.base.Throwables;
+import com.google.common.net.InetAddresses;
+import com.jayway.awaitility.Awaitility;
+import com.rabbitmq.client.BuiltinExchangeType;
+import com.rabbitmq.client.Channel;
+import com.rabbitmq.client.Connection;
+import com.rabbitmq.client.ConnectionFactory;
+import com.rabbitmq.client.GetResponse;
+
+public class AmqpRule extends ExternalResource {
+
+    private final SwarmGenericContainer rabbitMqContainer;
+    private final String exchangeName;
+    private final String routingKey;
+    private Channel channel;
+    private String queueName;
+    private Connection connection;
+    private String amqpUri;
+
+    public AmqpRule(SwarmGenericContainer rabbitMqContainer, String exchangeName, String routingKey) {
+        this.rabbitMqContainer = rabbitMqContainer;
+        this.exchangeName = exchangeName;
+        this.routingKey = routingKey;
+    }
+
+    @Override
+    protected void before() throws Throwable {
+        @SuppressWarnings("deprecation")
```

Comment:
Firstly @suppressWarnings does not compile on my eclipse (I have to put it at the method level). I guess it's again eclipse that suck.
Secondly, I have the exact same ugly line here: https://github.com/linagora/james-project/pull/559/files#diff-450812b86702bd99720bdc0a18b4703eR75.
I am afraid the day where we will figure out how to do that with the "new api" we will have a lot of line to modify.
If it's not too much too ask, could you refactor that.